hoprd/rest-api/src/session.rs

```diff
@@ -0,0 +1,131 @@
+        Self {
+            peer: value.destination,
+            path_options: value.path,
+            capabilities: value.capabilities,
+        }
+    }
+}
+
+#[serde_as]
+#[derive(Debug, Clone, Serialize, Deserialize, utoipa::ToSchema)]
+#[schema(example = json!({
+        "port": 5542,
+    }))]
+#[serde(rename_all = "camelCase")]
+pub(crate) struct SessionClientResponse {
+    pub port: u16,
+}
+
+/// Creates a new client session returing a dedicated session listening port.
+#[utoipa::path(
+        get,
+        path = const_format::formatcp!("{BASE_PATH}/session/client"),
+        request_body(
+            content = SessionClientRequest,
+            description = "Configuration of the client session to be created",
+            content_type = "application/json"),
+        responses(
+            (status = 200, description = "Successfully created a new client session", body = SessionClientResponse),
+            (status = 400, description = "Bad request.", body = ApiError),
+            (status = 401, description = "Invalid authorization token.", body = ApiError),
+            (status = 422, description = "Unknown failure", body = ApiError),
+            (status = 500, description = "Internal server error", body = ApiError)
+        ),
+        security(
+            ("api_token" = []),
+            ("bearer_token" = [])
+        ),
+        tag = "Messages"
+    )]
+pub(crate) async fn create_client(
+    State(state): State<Arc<InternalState>>,
+    Json(args): Json<SessionClientRequest>,
+) -> Result<impl IntoResponse, impl IntoResponse> {
+    let port = args.port.unwrap_or(0);
+    let data: SessionClientConfig = args.into();
+    let is_tcp_like = data.capabilities.contains(&SessionCapability::Retransmission)
+        || data.capabilities.contains(&SessionCapability::Segmentation);
+
+    if is_tcp_like {
+        let session = state.hopr.clone().connect_to(data).await.map_err(|e| {
+            (
+                StatusCode::INTERNAL_SERVER_ERROR,
+                ApiErrorStatus::UnknownFailure(e.to_string()),
+            )
+        })?;
+
+        let tcp_listener = TcpListener::bind("127.0.0.1:{port}").await.map_err(|e| {
+            (
+                StatusCode::INTERNAL_SERVER_ERROR,
+                ApiErrorStatus::UnknownFailure(format!("Failed to bind on 127.0.0.1:{port}: {e}")),
+            )
+        })?;
+
+        let port = tcp_listener
+            .local_addr()
+            .map_err(|e| {
+                (
+                    StatusCode::INTERNAL_SERVER_ERROR,
+                    ApiErrorStatus::UnknownFailure(format!("Failed to get the port number: {e}")),
+                )
+            })?
+            .port();
+
+        tokio::task::spawn(async move {
+            match tcp_listener
+            .accept()
+            .await {
+                Ok((mut tcp_stream, _sock_addr)) => match copy_bidirectional_with_sizes(&mut session.compat(), &mut tcp_stream, 1024, 1024).await {
```

Comment:
I would move out the 1024 literal into two constants - `IN_SIZE` and `OUT_SIZE` constants.
How does the inner mechanism flush the buffer when not full ? Let's say I send only a few bytes via TCP.

Also on we could set this to something closer to 1492.

Reply:
Set to 1456 and refactored out as const variables.